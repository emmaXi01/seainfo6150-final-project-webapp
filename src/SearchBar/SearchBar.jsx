import React, { useState } from "react";
import PropTypes from "prop-types";
import styles from "./SearchBar.module.css";

const SearchBar = (props) => {
  const [newText, setNewText] = useState('');

  const searchResult = () => {
    if(newText) {
      props.onSearch(newText);
      setNewText('');
    }
  };

  const onTextChange = (e) => {
    setNewText(e.target.value);
  };

  return (
    <div>
      <input className={styles.input} placeholder="Search for course" onChange={onTextChange} value={newText} />
      <button className={styles.searchButton} onClick={searchResult}>search</button>
    </div>
  );
};

SearchBar.propTypes = {
  onSearch: PropTypes.func.isRequired
};

export default SearchBar;
