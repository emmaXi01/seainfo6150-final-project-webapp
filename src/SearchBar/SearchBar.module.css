.input {
  border: 1px solid #989586;
  border-radius: 2px 0 0 2px; 
  height: 35px;
  padding: 0 10px 0 20px;
  /* font-family: Helvetica,sans-serif; */
  font-weight: 300;
  font-size: 1rem;
  box-shadow: none;
  /* margin-bottom: 0; */
  
  
}

.searchButton {
  border: 1px solid #4085d3;
  background-color: #4085d3;
  height: 35px;
  font-size: 1rem;
  /* font-size: 16px; */
  color: white;
  border: 0;
  padding: 0 10px 0 10px;
}

/* @media screen and (min-width: 400px) {
  .input, .searchButto{
    font-size: 0.85rem;

  }
} 

@media screen and (min-width: 600px) {
  .input, .searchButton{
    font-size: 1rem;
  }
} */