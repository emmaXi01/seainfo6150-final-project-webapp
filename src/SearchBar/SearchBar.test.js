import React from "react";
import { render } from "@testing-library/react";
import SearchBar from "./SearchBar.jsx";

describe("SearchBar tests", () => {
  it("renders the search bar component correctly", () => {
    const { container } = render(<SearchBar onSearch={()=>{}}/>);
    expect(container).toMatchSnapshot();
  });
});