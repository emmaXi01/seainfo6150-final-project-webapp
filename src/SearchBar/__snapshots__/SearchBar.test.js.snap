// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar tests renders the search bar component correctly 1`] = `
<div>
  <div>
    <input
      class="input"
      placeholder="Search for course"
      value=""
    />
    <button
      class="searchButton"
    >
      search
    </button>
  </div>
</div>
`;
